api/v0/proto/retrievability/types.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+package celestia.v0.retrievability.types;
+
+// TODO: oneof different blob versions
+message Blob {
+  // The blob data.
+  bytes data = 1;
+  // The signer of the blob.
+  optional bytes signer = 2;
+  // The blob commitment.
+  bytes commitment = 3;
+}
+
+// Blob with blob proof. Either the blob and the proof are optional, depending
+// on the options of the request.
+message BlobWithProof {
+  // The blob.
+  optional Blob blob = 1;
+  // The proof of blob inclusion or non-inclusion.
+  optional DataRootProof proof = 2;
+  // Index of the first share of the blob in the ODS.
+  uint64 index = 3;
+}
+
+// A single data root proof can span multiple rows.
+message DataRootProof {
+  // The type of the proof.
+  ProofType type = 1;
+  // The proof.
+  Proof proof = 2;
+}
+
+enum ProofType {
+  PROOF_TYPE_UNSPECIFIED = 0;
+  // Blob inclusion proof.
+  PROOF_TYPE_INCLUSION = 1;
+  // Blob non-inclusion proof.
+  PROOF_TYPE_NON_INCLUSION = 2;
+}
```

Comment:
What happened to Blob proofs (commitment inclusion)? They are different by nature and would be great to have explicit identifier for them

Reply:
Found an answer in https://github.com/celestiaorg/canonical-api/pull/2#discussion_r1988063442. However I still think there should be distinguish with explicit enum to allow library to identify which proving logic to apply. Otherwise user would need to convert generic proof to specific proofs on every occasion, which I think is poor UX

Reply:
LN can verify proofs, and it knows what kind of proof it's requesting.

Reply:
Thats true, but it would mean that there is implicit rule on which type of verification logic should be used against the proof depending on the name of the called RPC method. It is implicit logic, that we would need to specify in comments or other external doc. I would suggest us to make it explicit with enum, which will make different logic to be defined by value in proto contract. So instead of relying on users to carefully reading docs, there will be typed system to define verification behavior. It is more about typed vs untyped approach.